drivers/serial/uart_xlnx_ps.c

```diff
@@ -313,6 +385,446 @@ static void uart_xlnx_ps_poll_out(struct device *dev, unsigned char c)
+		break;
+	}
+
+	return true;
+}
+
+/**
+ * @brief Converts a data bit enum value to a Mode Register bit mask.
+ *
+ * Converts a value of an enumeration type provided by the driver
+ * framework for the configuration of the UART's data bit setting
+ * into a bit mask within the Mode Register.
+ *
+ * It is assumed that the Mode Register contents that are being
+ * modified within this function come with the bits modified by
+ * this function already masked out by the caller.
+ *
+ * @param mode_reg Pointer to the Mode Register contents to which
+ *                 the data bit configuration shall be added.
+ * @param databits Enumeration value to be converted to a bit mask.
+ *
+ * @return Indication of success or failure in case of an unspported
+ *         data bit configuration being provided by the caller.
+ */
+static inline bool uart_xlnx_ps_cfg2ll_databits(
+	u32_t *mode_reg,
+	enum uart_config_data_bits databits)
+{
+	/*
+	 * Translate the new data bit configuration to the mode register's
+	 * bits [2..1] (CHRL):
+	 *  0xb : 8 data bits
+	 *  10b : 7 data bits
+	 *  11b : 6 data bits
+	 */
+
+	switch (databits) {
+	case UART_CFG_DATA_BITS_5:
+	case UART_CFG_DATA_BITS_9:
+		/* Controller doesn't support 5 or 9 data bits */
+		return false;
+	default:
+	case UART_CFG_DATA_BITS_8:
+		*mode_reg |= XUARTPS_MR_CHARLEN_8_BIT;
+		break;
+	case UART_CFG_DATA_BITS_7:
+		*mode_reg |= XUARTPS_MR_CHARLEN_7_BIT;
+		break;
+	case UART_CFG_DATA_BITS_6:
+		*mode_reg |= XUARTPS_MR_CHARLEN_6_BIT;
+		break;
+	}
+
+	return true;
+}
+
+/**
+ * @brief Converts a flow control enum value to a Modem Control
+ *        Register bit mask.
+ *
+ * Converts a value of an enumeration type provided by the driver
+ * framework for the configuration of the UART's flow control
+ * setting into a bit mask within the Modem Control Register.
+ *
+ * It is assumed that the Modem Control Register contents that are
+ * being modified within this function come with the bits modified
+ * by this function already masked out by the caller.
+ *
+ * @param modemcr_reg Pointer to the Modem Control Register contents
+ *                    to which the flow control configuration shall
+ *                    be added.
+ * @param hwctrl Enumeration value to be converted to a bit mask.
+ *
+ * @return Indication of success or failure in case of an unspported
+ *         flow control configuration being provided by the caller.
+ */
+static inline bool uart_xlnx_ps_cfg2ll_hwctrl(
+	u32_t *modemcr_reg,
+	enum uart_config_flow_control hwctrl)
+{
+	/*
+	 * Translate the new flow control configuration to the modem
+	 * control register's bit [5] (FCM):
+	 *  0b : no flow control
+	 *  1b : RTS/CTS
+	 */
+
+	if (hwctrl == UART_CFG_FLOW_CTRL_RTS_CTS) {
+		*modemcr_reg |= XUARTPS_MODEMCR_FCM_RTS_CTS;
+	} else if (hwctrl ==  UART_CFG_FLOW_CTRL_NONE) {
+		*modemcr_reg |= XUARTPS_MODEMCR_FCM_NONE;
+	} else {
+		/* Only no flow control or RTS/CTS is supported. */
+		return false;
```

Comment:
Should this return `-ENOTSUP` instead of `false` since this statement is to catch all hwctrl values not supported?

Reply:
nvm I see that you handle the -ENOTSUP in ln 623 as part of _configure( ). Thanks! My comment about `1` and `0` still remains at this time.